frontend/kesaseteli/handler/src/hooks/application/useApplicationWithoutSsnFormField.ts

```diff
@@ -0,0 +1,80 @@
+/**
+ * TODO: YJDH-701, refactor to reduce code duplication, copied and modified from:
+ *       frontend/kesaseteli/employer/src/hooks/application/useApplicationFormField.ts
+ */
+import useGetApplicationWithoutSsnFormFieldLabel from 'kesaseteli/handler/hooks/application/useGetApplicationWithoutSsnFormFieldLabel';
+import type {
+  ApplicationWithoutSsn,
+  ApplicationWithoutSsnFieldName,
+  ApplicationWithoutSsnFieldPath,
+  ApplicationWithoutSsnFormField,
+  ApplicationWithoutSsnValue,
+} from 'kesaseteli/handler/types/application-without-ssn-types';
+import { useTranslation } from 'next-i18next';
+import { ErrorOption, FieldError, get, useFormContext } from 'react-hook-form';
+import { getLastValue } from 'shared/utils/array.utils';
+import { convertToUIDateFormat, parseDate } from 'shared/utils/date.utils';
+import { isDateObject } from 'shared/utils/type-guards';
+
+const useApplicationWithoutSsnFormField = <
+  V extends ApplicationWithoutSsnValue
+>(
+  id: ApplicationWithoutSsnFieldPath
+): ApplicationWithoutSsnFormField<V> => {
+  const { t } = useTranslation();
+  const {
+    control,
+    register,
+    getValues,
+    setValue,
+    watch,
+    trigger,
+    clearErrors,
+    formState,
+    setFocus,
+    setError: setErrorF,
+  } = useFormContext<ApplicationWithoutSsn>();
+  const fieldName = (getLastValue((id as string).split('.')) ??
+    '') as ApplicationWithoutSsnFieldName;
+  const defaultLabel = useGetApplicationWithoutSsnFormFieldLabel(fieldName);
+
+  const getError = (): FieldError | undefined =>
+    get(formState.errors, id) as FieldError | undefined;
```

Comment:
Trivial: All the rest are implemented in the return statement, so this is a bit inconsistent.

Reply:
It's because the `getError` function is being used in the definition of `hasError` and `getErrorText` functions.

Reply:
Not taken into account as there was a reason why it was different from the others